### PR TITLE
Remove normalization warning in `dq.wigner()` (not jit-compatible)

### PR DESCRIPTION
**Pull request by gautierronan**
Follow-up to https://github.com/dynamiqs/dynamiqs/pull/914 to fix the following MWE:
```python
import dynamiqs as dq
import jax

jax.jit(dq.wigner)(dq.coherent(8, 1.0))
```
cc @cyril87885

### CODE REVIEW

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Unfortunately this raises a warning whatever the state norm. 😕

```python
>>> psi = dq.basis(2, 0)
>>> dq.wigner(psi)
UserWarning: Argument state is not normalized (expected norm 1).
```

**Comment by gautierronan** [comment]
> Unfortunately this raises a warning whatever the state norm. 😕

Indeed, this is because both branches of the `lax.cond` are evaluated at trace time. Then I think we should eliminate this check altogether. It is better not to have the warning than to not be jit-compatible. 

Maybe a solution would be something similar to [Equinox runtime errors](https://docs.kidger.site/equinox/api/errors/#equinox.error_if), but it would have to be a runtime warning instead. Probably requires implementing something like [here](https://github.com/patrick-kidger/equinox/blob/8191b113df5d985720e86c0d6292bceb711cbe94/equinox/_errors.py#L283-L344).

**Comment by pierreguilmin** [comment]
I think we should either always warn or never, otherwise the user may wrongly assume that the norm is correct bc he is used to the warning mechanism which doesn't work sometimes.

**Comment by gautierronan** [comment]
Ok, then I removed the warning altogether. That's quite unfortunate :(

**Comment by pierreguilmin** [comment]
Yes it's a bit of a shame. 🥲

**Review by pierreguilmin — APPROVED** [review]

**Comment by pierreguilmin** [comment]
CC @CyrilDeloince, we have to revert #914. The warning is not compatible with the JAX JIT mechanism, and we couldn't find a workaround.
